content/blog/modern-text-features/index.Rmd

```diff
@@ -0,0 +1,370 @@
+    family = "Arial"
+  ) + 
+  expand_limits(y = c(0, 4))
+
+preview_devices(p, "rtl_example")
+```
+
+```{r, echo=FALSE, results='asis'}
+get_results("rtl_example")
+```
+
+Now, if you are not trained in reading any of the three languages above it may be hard to see what is right and what is wrong.
+You may, however, look at how the text in the code is rendered in the browser and compare that to the device rendering. If you do that you can see that the Hebrew script is rendered in the wrong direction for all the non-ragg devices (except Cairo on Linux).
+For the other scripts it is less obvious that the direction is wrong since the text glyphs are fundamentally different due to ligature substitutions.
+Still, by comparing to the browser rendering you can see that the same devices failing on the Hebrew script fail here as well.
+
+A point to make is that all of this "just works" in ragg.
+How is that?
+Shouldn't we have to indicate which kind of script we want to use?
+This is not necessary due to the genius of the Unicode standard which relates characters to specific scripts.
+The script, and by extension the layout, can thus simply be deduced from the provided string without needing to specify any additional information.
+
+The Cairo device on Linux handles this task well, as we have noted above.
+How come this works, but only on one OS (Cairo on macOS and Windows performs as bad as the other native devices)?
+Cairo is a fundamental library of many Linux distributions and is usually build on top of the Pango library which handles text layouting on Linux.
+It thus have access to OS level text rendering when used on Linux, but not on the other major platforms.
+The text rendering quality of the Cairo device on Linux is something that will be consistent through these examples and while it is very good, it is not fair to ask users to switch operating system to get modern font support.
+Even if you are already using Linux you'd have to accept that your output is not necessarily reproducible on other systems if you using the default device.
+
+### Bidirectional text
+
+In the case of a mix of scripts, most importantly a mix of scripts with different direction, the string needs to be split based on the bidirectional algorithm (also defined in the Unicode standard) and each script run should then be laid out individually and combined in the end.
+
+```{r, eval=FALSE}
+bidi_text <- "The Hebrew (עִברִית) script\nis right-to-left"
+
+p <- ggplot() + 
+  geom_text(
+    aes(x = 0, y = 0, label = bidi_text), 
+    family = "Arial"
+  )
+
+preview_devices(p, "bidi_example")
+```
+
+```{r, echo=FALSE, results='asis'}
+get_results("bidi_example")
+```
+
+It comes as no surprise that the devices that struggle with RtL scripts also fail when these are mixed with Left-to-Right (LtR), so the point here is mainly that ragg (and Cairo on Linux) supports this additional complication with no additional work on the user.
+
+### Scripts with special consideration
```

Comment:
I'd suggest dropping this section. If you think it needs more of a mention, I'd suggest a couple more sentences under "Advanced script support"